### PR TITLE
Document `action` RBAC action and application resource path. Relates to #3279

### PR DESCRIPTION
**Pull request by margueritepd**
Checklist:

* [x] this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note). 
* [ ] ~I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.~
    ✍🏻  This does not fully address #3279 since I don't have the knowledge to document each action.
* [ ] ~I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.~
* [ ] ~Does this PR require documentation updates?~
* [ ] ~I've updated documentation as required by this PR.~
* [ ] ~Optional. My organization is added to USERS.md.~
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] ~I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.~ not needed for docs
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

# Description

This PR is the result of me floundering around while making an RBAC change and is in no way based on any expertise of mine. I offer it as a starting point for more correct or detailed documentation on the Action and Application path used in RBAC. I welcome follow-up commits or comments to correct what I am saying.

### CODE REVIEW

File: docs/operator-manual/rbac.md [hunk 1]
```diff
@@ -30,7 +30,28 @@ Breaking down the permissions definition differs slightly between applications a
...
+
+Delete access to sub-resources of a project, such as a rollout or a pod, cannot
+be managed granularly. `<project-name>/<application-name>` grants access to all
+subresources of a project.
+
+#### The `action` action
+
+The `action` action corresponds to either built-in resource customizations defined
+[here](https://github.com/argoproj/argo-cd/search?q=filename%3Aaction.lua+path%3Aresource_customizations),
+or to [custom resource actions](resource_actions/index.md#custom-resource-actions) defined by you.
```
Comment (margueritepd) [inline docs/operator-manual/rbac.md:49]: Can you check that this is the right way to reference other parts of the documentation?

File: docs/operator-manual/rbac.md [hunk 2]
```diff
@@ -30,7 +30,28 @@ Breaking down the permissions definition differs slightly between applications a
...
 Resources: `clusters`, `projects`, `applications`, `repositories`, `certificates`, `accounts`, `gpgkeys`
 
-Actions: `get`, `create`, `update`, `delete`, `sync`, `override`, `action`
+Actions: `get`, `create`, `update`, `delete`, `sync`, `override`,
+`action/<path/to/action>`
+
+#### Application resources
+
+The resource path for application objects is of the form
+`<project-name>/<application-name>`.
```
Comment (margueritepd) [inline docs/operator-manual/rbac.md:39]: Can objects other than `application-name` be used? How would one reference a pod belonging to an application, for example?

File: docs/operator-manual/rbac.md [hunk 3]
```diff
@@ -30,7 +30,28 @@ Breaking down the permissions definition differs slightly between applications a
...
+
+Delete access to sub-resources of a project, such as a rollout or a pod, cannot
+be managed granularly. `<project-name>/<application-name>` grants access to all
+subresources of a project.
```
Comment (margueritepd) [inline docs/operator-manual/rbac.md:43]: I have no evidence that this is true. Please correct me and I will document it.

**Review by crenshaw-dev — COMMENTED** [review]
I spent an hour reading RBAC code, and I'll probably spend a couple more hours tomorrow. I will prevail and then leave a useful review, but for now, thanks for writing this up. It's very much needed. :-)

**Comment by crenshaw-dev** [comment]
Aaaaaaalrighty, after some distractions, I'm back on this. Thanks for your patience!

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Only small suggestions. I'm still uncertain about the more granular resource permissions and will keep reading source to be certain.

File: docs/operator-manual/rbac.md (hunk 2 above)
Comment (crenshaw-dev) [reply docs/operator-manual/rbac.md:39]: Based on [the policy validation regex](https://github.com/argoproj/argo-cd/blob/9bdf0bae2db6884f23b8ddb4b6dfc473b22ad6f3/pkg/apis/application/v1alpha1/types.go#L1600), I agree that only application names are allowed. No sub-resources.

**Comment by crenshaw-dev** [comment]
Ohp, I probably shoulda warned about the suggestions. If you apply them in the GitHub interface, the sign-off doesn't happen. :-/ But the check failure details show how to amend the commit.

**Comment by margueritepd** [comment]
@crenshaw-dev np I'll clean it up when the other argo devs chime in with their 2c and suggestions!

File: docs/operator-manual/rbac.md (hunk 3 above)
Comment (crenshaw-dev) [reply docs/operator-manual/rbac.md:43]: I now disagree with myself. I think this is the only change needed.

```suggestion
subresources of an application.
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
@margueritepd I'm now more confident in my understanding of the RBAC setup and have just one small change request. 🙂

**Comment by margueritepd** [comment]
@crenshaw-dev Sorry for the delay in addressing your comment! I lost track of this PR. I've incorporated your suggestion.

**Review by crenshaw-dev — APPROVED** [review]
LGTM! Thanks again!

**Pull request merged**